ortools/util/time_limit.h

```diff
@@ -529,7 +576,38 @@ inline double TimeLimit::ReadInstructionCounter() {
   return 0;
 }
 
+inline bool TimeLimit::Paused() {
+  if (external_boolean_as_pause_ != nullptr &&
+      external_boolean_as_pause_->load()) {
+    return true;
+  }
+  else {
+    return false;
+  }
+}
+
 inline bool TimeLimit::LimitReached() {
+  // LimitReached method handles the Paused state of the time_limit. 
+  // The pause can be interrupted by either unpausing or stopping the time_limit
+  // setting its respective external booleans.
+  // Note, external_boolean_as_limit must be registered prior to pausing the time_limit.
+  if (external_boolean_as_limit_ != nullptr) {
+    if (Paused()) {
+      int64 paused_time = absl::GetCurrentTimeNanos();
+      while (Paused() && !external_boolean_as_limit_->load()) {
```

Comment:
why not using a cond variable instead ?

Reply:
cond variable could be much better, but we also need to handle "stop" request there (to allow stop paused computation) - one possible way should be to use method "stopRequest" to "force resume paused computatïon when stop of computation is requested", but it is not usable when stop/pause booleans are "external" (so they are not changed via class methods only), so this way causes a lot of changes across project (we tried this, but without success :( ).